### PR TITLE
[FIX] website_event: use correct subtype when (un)publishing

### PR DESCRIPTION
**Pull request by nda-odoo**
`is_published` has been renamed in `website_published` in e3d2b01 but not in `_track_subtype`().

**steps to reproduce:**
- add a follower to an event, select the two event subtypes on this follower
- check this event on the website and publish/unpublish it
- check chatter, tracking values have been created
- no message have been sent
- 
**before this commit:**
- subtypes `mt_event_published` and `mt_event_unpublished` are not used

**after this commit:**
- a message is sent when an event is (un)published

opw-3582596

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website_event/models/event_event.py [hunk 1]
```diff
@@ -400,9 +400,9 @@ def _google_map_link(self, zoom=8):
 
     def _track_subtype(self, init_values):
         self.ensure_one()
-        if 'is_published' in init_values and self.is_published:
+        if 'website_published' in init_values and self.is_published:
             return self.env.ref('website_event.mt_event_published')
```
Comment (tde-banana-odoo) [inline addons/website_event/models/event_event.py:404]: Maybe we could take the opportunity to use raise_if_not_found=False and use the "note" subtype if it has been removed, to be defensive.

File: addons/website_event/models/event_event.py [hunk 2]
```diff
@@ -400,9 +400,9 @@ def _google_map_link(self, zoom=8):
 
     def _track_subtype(self, init_values):
         self.ensure_one()
-        if 'is_published' in init_values and self.is_published:
+        if 'website_published' in init_values and self.is_published:
```
Comment (tde-banana-odoo) [inline addons/website_event/models/event_event.py:403]: We could be include, and hook on both fields maybe ? Like
```suggestion
        if init_values.keys() & {'is_published', 'website_published'} and self.is_published:
```

File: addons/website_event/tests/test_website_event.py [hunk 3]
```diff
@@ -119,3 +119,34 @@ def test_sitemap(self):
         self.assertTrue('/event/event-0' in resp.text, 'Published events must be present in the sitemap')
         self.assertTrue('/event/event-1' in resp.text, 'Published events must be present in the sitemap')
         self.assertFalse('/event/event-2' in resp.text, 'Unpublished events must not be present in the sitemap')
+
+class TestMail(MailCommon):
```
Comment (tde-banana-odoo) [inline addons/website_event/tests/test_website_event.py:123]: Could probably belong to its own test file, like 'test_event_mail.py' with mail related stuff we might want to add :)

File: addons/website_event/tests/test_website_event.py [hunk 4]
```diff
@@ -119,3 +119,34 @@ def test_sitemap(self):
...
+        published_subtype = self.env.ref('website_event.mt_event_published')
+        unpublished_subtype = self.env.ref('website_event.mt_event_unpublished')
+        event = self.env['event.event'].create({
+            'name': 'Event',
+            'date_begin': datetime.today() - timedelta(days=1),
+            'date_end': datetime.today() + timedelta(days=1),
+        })
+        self.flush_tracking()
+
+        follower = self.env['res.users'].create({
```
Comment (tde-banana-odoo) [inline addons/website_event/tests/test_website_event.py:136]: There are already test users in mailcommon (notably a standard employee), try to reuse existing data

**Review by tde-banana-odoo — COMMENTED** [review]
Thanks for the fix, some remarks on my side :)

File: addons/website_event/models/event_event.py (hunk 1 above)
Comment (nda-odoo) [reply addons/website_event/models/event_event.py:404]: Ok for the `raise_if_not_found` . But I'm not sure about the `note` . If the custom subtype has been deleted we want to default to the standard behavior no? So let `_track_subtype` return `False` and in the end use a log message.

But you have the final word and it's a simple change so say the word and i will add the "note" fallback.

**Comment by nda-odoo** [comment]
Hello @tde-banana-odoo , I pushed the changes. Is it ok for you?

File: addons/website_event/models/event_event.py [hunk 5]
```diff
@@ -400,10 +400,11 @@ def _google_map_link(self, zoom=8):
...
     def _track_subtype(self, init_values):
         self.ensure_one()
-        if 'is_published' in init_values and self.is_published:
-            return self.env.ref('website_event.mt_event_published')
-        elif 'is_published' in init_values and not self.is_published:
-            return self.env.ref('website_event.mt_event_unpublished')
+        if init_values.keys() & {'is_published', 'website_published'}:
+            if self.is_published:
+                return self.env.ref('website_event.mt_event_published', raise_if_not_found=False)
+            else:
```
Comment (tde-banana-odoo) [inline addons/website_event/models/event_event.py:406]: Note that the else is not necessary, as you return in the "if"

**Review by tde-banana-odoo — COMMENTED** [review]
Small code details :)

**Comment by nda-odoo** [comment]
You're right. We'll never achieve -20% in term of line of codes if we keep adding useless lines ;-)

**Comment by tde-banana-odoo** [comment]
@robodoo r+

**Pull request closed without merging**